### PR TITLE
Infra: improve generator initial boilerplate and style override mechanism

### PR DESCRIPTION
**Pull request by shlomitc**
Following this PR: https://github.com/wix/wix-ui-tpa/pull/456

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_463/

File: .wuf/generator/templates/desktop/Component/Component.st.css [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
 }
 
-/*Overrides*/
+/* Overrides - deceleration of overridable parameters by the consumer */
 :vars {
-    /*Deceleration of overridable placeholders*/
+  /*
+    Main background color
+  */
+  BgColor: --overridable;
```
Comment (jonathanadler) [inline .wuf/generator/templates/desktop/Component/Component.st.css:16]: Do you maybe want to use the autodocs Stylable plugin "syntax" of adding defaults?
https://github.com/wix/wix-ui-tpa/blob/48aa71d1a6cd07f139561d4b559a80754e72977c/src/components/TextArea/TextArea.st.css#L51

And maybe we don't even need to split the `defaults` and the `overridables`?
Just have it:
```css
  /*
    Main background color
    @default color-1
  */
  BgColor: color(color-1);
```

File: .wuf/generator/templates/desktop/Component/Component.st.css [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
 }
 
 .root {
-    /*css goes here*/
+  background-color: value(defaultBgColor);
 }
+
+.styleParams { 
+  background-color: value(bgColor);
+}
```
Comment (jonathanadler) [inline .wuf/generator/templates/desktop/Component/Component.st.css:30]: even though the TPA's use the `styleParams` to refer to the variables coming from the settings
I'm not sure that in this context it's completely understood

I mean, if this wasn't `css` I think I would go with `applyStyles` or something like that
but then `.applyStyles{ ... }` is a bit weird

idk, mainly thinking out loud

**Comment by jonathanadler** [comment]
Also, we probably need to think of how to document this, and let developers know that this is the "new" way to override styles.
We'll maybe need some indicator for each component

File: .wuf/generator/templates/desktop/Component/Component.st.css (hunk 2 above)
Comment (shlomitc) [reply .wuf/generator/templates/desktop/Component/Component.st.css:30]: Other options can be paramsOverrides or overrideParams.

I want a name that is short yet explicit enough to explain the intent.

**Comment by shlomitc** [comment]
> Also, we probably need to think of how to document this, and let developers know that this is the "new" way to override styles.
> We'll maybe need some indicator for each component

We can have a single doc (reflected in the storybook as well) that explains the pattern and point to it from each story page

File: .wuf/generator/templates/desktop/Component/Component.st.css (hunk 1 above)
Comment (shlomitc) [reply .wuf/generator/templates/desktop/Component/Component.st.css:16]: I'll need to experiment it further so I'm keeping it out of this current PR

**Review by jonathanadler — APPROVED** [review]

**Comment by jonathanadler** [comment]
> > Also, we probably need to think of how to document this, and let developers know that this is the "new" way to override styles.
> > We'll maybe need some indicator for each component
> 
> We can have a single doc (reflected in the storybook as well) that explains the pattern and point to it from each story page

Should this be part of this PR?

**Comment by shlomitc** [comment]
image

**Review by jonathanadler — APPROVED** [review]
Inspiring 👏 👏 👏

**Comment by shlomitc** [comment]
merged as part of #467

**Pull request closed without merging**